File: tests/indexing/parsing/lang/fixtures/typescript/Loops.ts
function joinParts(parts: string[]): string {
  let out = "";
  for (const part of parts) {
    out = out + part;
    out = out + ",";
  }
  return out;
}

function fib(count: number): number {
  let a = 0;
  let b = 1;
  for (let i = 0; i < count; i++) {
    const next = a + b;
    a = b;
    b = next;
  }
  return a;
}

function digits(value: number): number {
  let steps = 0;
  while (value > 0) {
    value = Math.floor(value / 10);
    steps = steps + 1;
  }
  return steps;
}

function retry(task: () => void): number {
  let attempts = 0;
  do {
    task();
    attempts = attempts + 1;
  } while (attempts < 3);
  return attempts;
}

File: tests/indexing/parsing/lang/fixtures/go/Loops.go
package example

func join(parts []string) string {
	out := ""
	for _, part := range parts {
		out = out + part
		out = out + ","
	}
	return out
}

func fib(count int) int {
	a, b := 0, 1
	for i := 0; i < count; i++ {
		next := a + b
		a = b
		b = next
	}
	return a
}

func drain(value int) int {
	steps := 0
	for value > 0 {
		value = value / 10
		steps = steps + 1
	}
	return steps
}
